tests/test_vertex_worker.py

```diff
@@ -205,3 +208,59 @@ async def test_cancelled_worker_run(self, flow_run, job_config):
             assert result == VertexAIWorkerResult(
                 status_code=1, identifier=job_display_name
             )
+
+    async def test_kill_infrastructure(self, flow_run, job_config):
+        mock = job_config.credentials.job_service_client.create_custom_job
+        # the CancelCustomJobRequest class seems to reject a MagicMock value
+        # so here, we'll use a SimpleNamespace as the mocked return values
+        mock.return_value = SimpleNamespace(
```

Comment:
im not fully understanding why, but returning a plain MagicMock here with the correct `name` and `state` keys would cause the test to fail, somewhere in the `CancelCustomJobRequest` class

```
self = <[AttributeError('Unknown field for CancelCustomJobRequest: _pb') raised in repr()] CancelCustomJobRequest object at 0x168bb4490>
mapping = {'name': <MagicMock name='mock_name.name' id='5749201248'>}, ignore_unknown_fields = False, kwargs = {'name': <MagicMock name='mock_name.name' id='5749201248'>}
params = {'name': <MagicMock name='mock_name.name' id='5749201248'>}, marshal = <proto.marshal.marshal.Marshal object at 0x106f4c790>, key = 'name'
value = <MagicMock name='mock_name.name' id='5749201248'>, pb_value = <MagicMock name='mock_name.name' id='5749201248'>
```

so, here i just use a `SimpleNamespace` to return arbitrary key values that can be access via dot notation